### DISCUSSION TITLE
On the dev server (yarn dev). I want to run my own script to generate some code when files have changed before nuxt rebuild the project. How can I do it?

### DISCUSSION THREAD

**Discussion by heart** [nuxt/nuxt#16816] [Questions]
On the dev server (yarn dev). 
I want to run my own shell script (or js function) 
to generate layouts folder
when files have changed before Nuxt rebuilds the project. 

How can I do it?

**Comment by sevillaarvin** [comment]
Not sure what you mean, if you want to create a folder when you execute `yarn dev` then you can use the npm pre hook:

https://docs.npmjs.com/cli/v8/using-npm/scripts#pre--post-scripts

Otherwise, I think you will have to use these hooks (I haven't tried them yet)

https://v3.nuxtjs.org/guide/going-further/hooks/
